ehrapy/api/tools/nlp/_translators.py

```diff
@@ -0,0 +1,252 @@
+                target_column = self.translator.translate_text(column, target_lang=target_language).text
+            if not inplace:
+                f"{target_column}_{target_language}"
+
+            adata.obs[target_column] = adata.obs[column].apply(
+                lambda text: self.translator.translate_text(text, target_lang=target_language).text
+            )
+
+    def translate_var_column(
+        self,
+        adata: AnnData,
+        target_language: str,
+        columns=Union[str, List],
+        translate_column_name: bool = False,
+        inplace: bool = False,
+    ) -> None:
+        """Translates a single var column and optionally replaces the original values
+
+        Args:
+            adata: :class:`~anndata.AnnData` object containing the obs column to translate
+            target_language: The target language to translate into, e.g. EN-US
+            columns: The columns to translate. Can be either a single column (str) or a list of columns
+            translate_column_name: Whether to translate the column name itself
+            inplace: Whether to replace the obs values or add a new obs column
+        """
+        if isinstance(columns, str):
+            columns = [columns]
+
+        for column in columns:
+            # as of Pandas 1.1.0 the default for new string column is still 'object'
+            if adata.var[column].dtype != str and adata.var[column].dtype != object:
+                raise ValueError("Attempted to translate column {column} which does not contain only strings.")
+            target_column = column
+            if translate_column_name:
+                target_column = self.translator.translate_text(column, target_lang=target_language).text
+            if not inplace:
+                f"{target_column}_{target_language}"
+
+            adata.var[target_column] = adata.var[column].apply(
+                lambda text: self.translator.translate_text(text, target_lang=target_language).text
+            )
+
+    def translate_X_column(
+        self,
+        adata: AnnData,
+        target_language: str,
+        columns=Union[str, List],
+        translate_column_name: bool = False,
+    ) -> None:
+        """Translates a X column into the target language in place.
+
+        Note that the translation of a column in X is **always** in place.
+
+        Args:
+            adata: :class:`~anndata.AnnData` object containing the var column to translate
+            target_language: The target language to translate into, e.g. EN-US
+            columns: The columns to translate. Can be either a single column (str) or a list of columns
+            translate_column_name: Whether to translate the column name itself (only translates var_names, not var)
+        """
+        if isinstance(columns, str):
+            columns = [columns]
+
+        for column in columns:
+            index: int = get_column_indices(adata, column)[0]
+            column_values = get_column_values(adata, index)
```

Comment:
In `get_column_indices` and `get_column_values` you already loop over (possibly) all var names. It would be more performant to actually move (at least the index function) outside of this loop and passing the `columns` parameter as list (I think it can handle it, what I remember). So this loop only runs once like O(n) and not (worst case, O(n^2)).